core/server/src/main/java/alluxio/master/file/meta/InodeDirectory.java

```diff
@@ -210,7 +230,8 @@ public static InodeDirectory fromJournalEntry(InodeDirectoryEntry entry) {
             .setPinned(entry.getPinned())
             .setLastModificationTimeMs(entry.getLastModificationTimeMs())
             .setPermissionStatus(permissionStatus)
-            .setMountPoint(entry.getMountPoint());
+            .setMountPoint(entry.getMountPoint())
+            .setDirectChildrenLoaded(entry.getDirectChildrenLoaded());
```

Comment:
It is an optional field. So if the entry is old, this will just return the default which is false.  An optional field that is not set doesn't affect serialized proto